src/ripple/rpc/handlers/GetAggregatePrice.cpp

```diff
@@ -170,21 +170,49 @@ doGetAggregatePrice(RPC::JsonContext& context)
     if (!params.isMember(jss::quote_asset))
         return RPC::missing_field_error(jss::quote_asset);
 
+    // Lambda to validate uint type
+    // support positive int, uint, and a number represented as a string
+    auto validUInt = [](Json::Value const& params,
+                        Json::StaticString const& field) {
+        auto const& jv = params[field];
+        std::uint32_t v;
+        return jv.isUInt() || (jv.isInt() && jv.asInt() >= 0) ||
+            (jv.isString() && beast::lexicalCastChecked(v, jv.asString()));
+    };
+
     // Lambda to get `trim` and `time_threshold` fields. If the field
     // is not included in the input then a default value is returned.
-    auto getField = [&params](
+    auto getField = [&params, &validUInt](
                         Json::StaticString const& field,
                         unsigned int def =
                             0) -> std::variant<std::uint32_t, error_code_i> {
         if (params.isMember(field))
         {
-            if (!params[field].isConvertibleTo(Json::ValueType::uintValue))
-                return rpcORACLE_MALFORMED;
+            if (!validUInt(params, field))
```

Comment:
What is the difference between these two lines of code?

Reply:
If I'm reading the code correctly, it's that the old version accepted booleans and floats and would just convert them?

Reply:
`isConvertible` is 
>      return (other == nullValue && value_.uint_ == 0) ||
>                (other == intValue && value_.uint_ <= (unsigned)maxInt) ||
>               other == uintValue || other == realValue ||
>               other == stringValue || other == booleanValue;
which means it can convert a wider range of other type. For instance, it'll convert 1.2 to 1, which is invalid in this case.

Reply:
> If I'm reading the code correctly, it's that the old version accepted booleans and floats and would just convert them?

Correct.